agents-api/agents_api/queries/executions/get_execution.py

```diff
@@ -38,14 +38,20 @@
     END AS error,
     coalesce(lt.output, '{}'::jsonb) AS output,
     lt.current_step,
+    tc.count as transition_count,
     lt.next_step,
     lt.step_label,
     lt.task_token,
     lt.metadata AS transition_metadata
 FROM
     executions e
     LEFT JOIN transitions lt ON e.execution_id = lt.execution_id
-    WHERE e.execution_id = $1
+    CROSS JOIN LATERAL (
+        SELECT COUNT(*) as count
+        FROM transitions t
+        WHERE t.execution_id = e.execution_id
+    ) tc
+WHERE e.execution_id = $1
 ORDER BY lt.created_at DESC
 LIMIT 1;
```

Comment:
@ellipsis-dev change this to:

```sql
FROM executions e
CROSS JOIN LATERAL (
    SELECT COUNT(*) AS count
    FROM transitions t
    WHERE t.execution_id = e.execution_id
) tc
LEFT JOIN LATERAL (
    SELECT next_step, step_label, task_token, metadata
    FROM transitions t
    WHERE t.execution_id = e.execution_id
    ORDER BY created_at DESC
    LIMIT 1
) lt ON true
WHERE e.execution_id = $1;
```

since we dont need to join on anything other than `t.execution_id = e.execution_id`, the LATERAL left join is much more efficient